### DISCUSSION TITLE
`kubebuilder create api` failed

### DISCUSSION THREAD

**Discussion by bestgopher** [kubernetes-sigs/kubebuilder#2526] [Support] [answered]
I am new to kubebuilder. When I follow the book add a new api by `kubebuilder create api` command, an error raised.
<img width="949" alt="image" src="https://user-images.githubusercontent.com/36840497/156880662-df97fcdf-0ac5-4e65-932c-1e3835f6b618.png">
<img width="303" alt="image" src="https://user-images.githubusercontent.com/36840497/156880690-a05690c7-00b2-4d0a-a7d3-3aa6c3d12068.png">

**Comment by camilamacedo86** [comment, marked as answer]
H @bestgopher, 

It seems that you were using an old version release. However, the problem here is that we are looking for the controller gen in the bin and you have it globally.  You probably also customize/change the Makefile so it is no longer working.

So you can check here the current implementation ( from the latest release 3.5.0 ): https://github.com/kubernetes-sigs/kubebuilder/blob/26d12ab1134964dbbc3f68877ebe9cf6314e926a/testdata/project-v3/Makefile#L125-L128

Regards customizations/changes see: 

Be aware of: 

<img width="865" alt="Screenshot 2022-07-27 at 11 39 06" src="https://user-images.githubusercontent.com/7708031/181227617-88adcfd3-aad7-4236-a5d3-03cfc0bd3f20.png">

From: https://book.kubebuilder.io/migration/v2vsv3.html#project-customizations

However, if you still facing any problems please open an issue in the repo and provide us more detailed info about what is the version used for example. Also, please feel free to contribute to the project if you think that we could improve this target.

**Comment by camilamacedo86** [comment]
Closing since it is answered.
